buildSrc/subprojects/parser/src/main/java/org/apache/calcite/buildtools/parser/DialectGenerate.java

```diff
@@ -120,10 +124,152 @@ public void unparseReservedKeywords(ExtractedData extractedData) {
       tokens.add(tokenBuilder.toString());
     }
     stringBuilder.append(String.join("| ", tokens));
-    stringBuilder.append("}\n");
+    stringBuilder.append("}");
     extractedData.tokenAssignments.add(stringBuilder.toString());
   }
 
+  /**
+   * Partitions {@code extractedData.nonReservedKeywords} amongst multiple
+   * functions and creates one function that calls all of them of the form:
+   *
+   * // Auto generated.
+   * void NonReservedKeyword():
+   * {
+   * }
+   * {
+   *     (
+   *         NonReservedKeyword0of<numPartitions>()
+   *     |   NonReservedKeyword1of<numPartitions>()
+   *     ...
+   *     )
+   *     { return unquotedIdentifier(); }
+   * }
+   *
+   * If {@code extractedData.nonReservedKeywords} is empty, the function only contains
```

Comment:
I suppose it would, I did this to ensure that this function always exists regardless of if nonReservedKeywords are specified. This is because it is referenced somewhere in parsing/parserImpls.ftl so not having it would prevent compiling. I'm not particularity attached to this solution, do you have a better approach?

Reply:
If there are no keywords, then you might want to just insert the <EOF> token here. So that would look like:

`void NonReservedKeyword():
{
}
{
  <EOF>
  { return unquotedIdentifier(); }
}
`

We should also add a test for this scenario of no nonreserved keywords.

Reply:
Will do, there is already a test but I'll modify it to include this.

Reply:
I meant a test that actually exercised the parsing codepath where we have no nonreserved keywords. For example, in a *ParserTest.java file for a dialect with no non-reserved keywords.

This might be overkill and unnecessary, but just wanted to clarify what was meant.

Reply:
Ah ok I see what you mean. I feel like the existing unit test covers this scenario quite well but I can also create a new structure just to test this if you want.